CHANGELOG.md

```diff
@@ -14,6 +14,8 @@ All notable changes to this project will be documented in this file.
 - Fields `region`, `endpoint` and `credentials` added to the `dynamodb` configuration section of the `aws_kinesis` input. (@jreyeshdez, @mihaitodor)
 - Field `isolation_level` added to `kafka`, `kafka_franz`, `ockam_kafka`, `redpanda`, `redpanda_common`, and `redpanda_migrator` inputs. (@rockwotj)
 - New `cohere_rerank` processor to rerank documents in RAG pipelines using Cohere. (@rockwotj)
+- Field `topic_prefix` added to the `redpanda_migrator` output. (@mihaitodor)
+- Field `offset_topic_prefix` added to the `redpanda_migrator_offsets` output. (@mihaitodor)
```

Comment:
I decided to go with the easy route and only support prefixes. It's still a bit awkward, since the `topic` field on the `redpanda_migrator` output allows users to configure any interpolated value, but if they do that, it's no longer possible to know what the input topic was, so we can't fetch its config. Might need to revise this in the future (or maybe hardcode the `topic` field to `${! @kafka_topic }` and prevent users from changing it...

Reply:
+1 to hard coding if that breaks things.

Reply:
I thought about it and the `redpanda_migrator_bundle` output template will reject configs which attempt to set the `topic` field for `redpanda_migrator`, since that's set by the template to the `@kafka_topic` metadata. In general, I don't expect people to use the `redpanda_migrator` output directly, so would rather not make the `FranzWriter` code overly-complex for now.